Update master branch from Train to Xena

msherman64 opened the pull request:
This PR contains the needed changes to move us from Train to Xena on the master branch. It also contains a number of bug-fixes and convenience features, some of which could be backported to train.

Backportable changes:
- Defaults for allowing a minimal dev site (disable tls, disable federation, set ironic inspection network and site_name)
- ensure that variables target the deploy host, in particular the path for templating the admin-openrc.sh
- split post-deploy into roles and playbooks for easier maintenance
- Allow deployment of KVM compute node


Breaking Changes:
New passwords must be set, use the `kolla-mergepwd` tool to pull in new values from the site-config.example

The hosts file has major changes, in particular the replacement of the HAProxy with Loadbalancer groups, addition of "common" group mostly replacing baremetal, and addition of a "defaults" group for the deploy host fix above. Back up your ansible hosts file, and then copy the one from site-config.example into its place, and update the primary groups as usual. Take care to review any customizations you may have done, and see if they are still needed.

Mark-Powers committed:
Update policies to YAML, removed deprecated policies

* Deletes neutron-fwaas policies
* Deletes loadbalancer policies
* Deletes policies moved out of neutron
* Deletes plural resource properties that were unused
* Policy `os_compute_api:os-hypervisors` is made granular

zhenz committed:
changes for filesystem

- add a filesystem role and include in post deploy
- filesystem role will create the default share type and ganesha
  external network
- add manila conf file and policy file
- add horizon custom local settings for manila

msherman64 committed:
update requirements from train to xena

Pin ansible to versions compatible with the xena branch of kolla-ansible
In addition, remove unused dependencies

msherman64 committed:
fix vault invocation for ansible 4.x

msherman64 committed:
[Hack] set PASSWORDS_FILE to pass prechecks

kolla-ansible has added a precheck for items in the passwords file with
empty values. Likely due to either our invocation/modifications of
kolla-ansible, setting `--passwords` does not successfully set the value
of PASSWORDS_FILE.

In addition, this check is passing trivially! The task is scaning an
encrypted file, not the variables loaded from it. Therefore it is a noop
in the case where the passwords are encrypted by vault.

msherman64 committed:
set default for site_name

msherman64 committed:
enable deployment of kvm compute node

This enables deployment of a nova-compute worker node to run KVM
instances. Add the desired host to the inventory/hosts file to deploy.

The post-deploy roles download some example images, and flavors. As
this feature is still in beta for Chameleon, they are set to admin-only
to avoid user confusion.

msherman64 committed:
ensure that defaults variables affect deploy host

Fix case where the deploy host was excluded from values defined in defaults.yml
To do so, create a new group, `defaults`, and include both `deployment` and `common`

`deployment` is not included in `common` as the user may be unprivileged on the deploy
host, and we don't want to run other roles there (e.g. fluentd, logging, etc)

msherman64 committed:
[neutron, ovs] split ml2 config for xena compat

In xena, openvswitch is configured via its own openvswitch_agent.ini,
and the fields ignored in ml2_conf.ini

msherman64 committed:
[ironic] update configuration for xena

Upgrade note! Make sure to set interfaces on baremetal nodes BEFORE
applying this, or any other change that removes `enabled_` interfaces.

This can be done via
`openstack baremetal node unset --deploy-interface <node>`
And similar commands. For the xena upgrade, the `iscsi` deploy interface
is removed, not just deprecated.

If upgrading an old deployment, best practice is to set the new defaults
in the ironic conductor, then reset all the nodes' defaults, and only
then remove the interface from `enabled_`. If this cannot be done, e.g.
you updated from Train to Xena in one go, you may have to update this
in the ironic database directly.

Changes for xena compatibiltiy:
- renamed pxe_append_params to kernel_append_params

General improvements:
- set ipxe as default boot method
- allow setting default deploy kernel and ramdisk
- make other default interface explicit

msherman64 committed:
don't enable tls by default

don't set this value. Instead, reuse the behavior from kolla's group_vars:
```
kolla_enable_tls_internal: "no"
kolla_enable_tls_external: "{{ kolla_enable_tls_internal if kolla_same_external_internal_vip | bool else 'no' }}"
```

this will allow use of single api endpoint for dev sites, and match upstream
docs more closely.

Sites will need to manually set `kolla_enable_tls_external: yes`

msherman64 committed:
disable neutron FWaaS by default

disable firewall as a service by default. It has been long unsupported.

msherman64 committed:
Set default for ironic inspection network

prechecks won't pass without an ironic-inspection network defined.
However, this is currently unused, and really only useful for the
standalone ironic case, as we define the ironic provisioning,
cleaning, and inspection networks in neutron.

msherman64 committed:
use upstream keystone IDP federation method

This adds static IDP metadata files for keystone, corresponding to
auth.chameleoncloud.org and auth.dev.chameleoncloud.org

By default, federation is enabled if keystone_idp_client_id is defined,
 and not null or false (truthy)

The `identity_provider_key` is set somewhat messily to correspond to the correct
hostname, so that the certificate can be loaded.

This needs a fair amount of cleanup, as upstream didn't use our (much cleaner)
way of fetching this data from the `.well_known` IDP endpoint.

msherman64 committed:
[ironic,neutron] split post-deploy into roles

split the post-deploy configuration into roles, especially to simplify
configuration of networking and baremetal

msherman64 committed:
[neutron] enable network_segment_range plugin

msherman64 committed:
reuse playbooks for post-deploy

To encourage code reuse, import existing playbooks rather than defining
the roles twice. Due to ansible path magic, we need to copy the whole
playbooks dir into the venv.

msherman64 committed:
update ansible-galaxy role installation for 4.x

Ansible-galaxy supports a new format to distingush roles from
collections.

msherman64 committed:
update inventory/hosts file for xena update

haroxy group has been replaced by loadbalancer group. See:
https://docs.openstack.org/releasenotes/kolla-ansible/xena.html#relnotes-13-0-0-stable-xena-upgrade-notes

common group has been added, replaces baremetal group. Also ensure that defaults.yml
is applied to this new group.

Mark-Powers commented on kolla/defaults.yml:
```diff
@@ -196,7 +205,7 @@ enable_jupyterhub_external_frontend: "{{ enable_jupyterhub_external }}"
 enable_neutron: yes
 enable_openvswitch: yes
-enable_neutron_fwaas: yes
+enable_neutron_fwaas: no
```
is `fwaas` supported anymore? I don't think it has a xena release. If not, can this be removed entirely?

Mark-Powers commented on kolla/defaults.yml:
```diff
@@ -129,18 +130,22 @@ horizon_regions:
     blazar_database_password: "{{ blazar_database_password }}"
 
 # Keystone IdP federation
+identity_provider_key: "{{ 'EABNO4wxhX0ZOryErTXnWPUkCAcWxTFHwZ6f_zjDHLg' if keycloak_hostname == 'auth.dev.chameleoncloud.org' else '02GplnOEWB4gwqrZDyyiaNdIrhgdinNnObXcPHy2RTo' }}"
```
Is it possible to convert these to friendlier names?

zhenz committed:
[Xena] clean up glance node_custom_config

Remove special config needed for multi-store in Train, as it is no longer needed in Xena.

To enable multi-store support at a site, simply enable the relevant store in the site-config, for example,
`glance_backend_swift: yes`

msherman64 committed:
Fix vendordata post-deploy roles

This fixes two issues.
1. the syntax to become a specific user is no longer valid, but is
   not needed, as the role uses become where necessary
2. Ensure the docker image is downloaded. The controller node may not be
   logged into the chameleon docker registry, in which case the systemd
   unit will fail to download the iamge or start.

msherman64 committed:
ensure image tools can download container

Log into the registry to allow image pulling.

msherman64 committed:
Ensure post-deploy roles succeed

This fixes several issues with post-deploy. In particular, kolla-ansible
is very sensitive to the paths for roles and playbooks, to the extent
that we copy the playbooks and/or post-deploy yml files into the venv.

However, using "import_playbook" is currently proving intractible.

Therefore, this reverts to the method of having a single post-deploy
playbook calling roles, but also maintaining separate playbooks to
exercise the roles outside of post-deploy.

msherman64 committed:
fixup ironic gateway interface roles

We can no longer call michaelrigart.interfaces as a module, and must
instead reference it as a role. This usage matches the docs for the role
and likely changed due to a role or ansible version update.

msherman64 committed:
Remove asertions from post_networking role

Allow the networking role to pass even if public network is not defined.
It will be largely a no-op, but allows the rest of post-deploy to
complete.

This is done in preparation for the addtion of  a separate "smoke tests"
playbook, to validate the status of the deployment.

msherman64 committed:
use become to access docker socket

msherman64 committed:
fix docker access for sharednet

msherman64 committed:
ensure ironic network can be created

the os_network module requires a project to be specified.
The project name is not included in the default openstack_auth var

msherman64 committed:
and also for ironic subnet

msherman64 committed:
fix syntax for vendordata handler

msherman64 committed:
ensure role can run with reduced privileges

Kolla_toolbox and docker roles need `become` if the user is not a member of the docker group.
We also need to authenticate to the registry.

msherman64 committed:
add missing become to role

msherman64 committed:
document possible options in defaults.yml (#199)

Defaults.yml has been fairly inscrutable in the past. This adds the following:

- A very minimal defaults.yml for evaluation
- An annotated defaults.yml to show what configuration options are valid

The comments document where to find defaults, how to modify them, and the precedence in which different files override each other.

zhenz committed:
Merge pull request #194 from ChameleonCloud/filesystem-latest

changes for filesystem

msherman64 committed:
Update default reviewers

msherman64 commented:
This is on hold until all associate sites are upgraded, as changing what the "master" branch does will affect existing sites.

msherman64 committed:
encode root pwd for ironic pxe agent

allow console access to ironic pxe agent. Root password must be
md5 encoded, then have "$" escaped, and finally enclosed in double
quotes. Details at:
https://docs.openstack.org/diskimage-builder/latest/elements/dynamic-login/README.html

At the serial console, log in with username: `root`, and password
from the vault encrypted passwords.yml, `ironic_pxe_root_password`

msherman64 committed:
allow customization of serial console speed

set serial console speed to 115200 by default. this significantly
speeds up the boot process for the ironic pxe agent

allow override if necessary due to incompatible hardware

zhenz committed:
[glance] clean up config overwrites
